src/lib/libemval.js

```diff
@@ -109,6 +110,85 @@ var LibraryEmVal = {
     for (var i = 0; i < view.length; i++) a[i] = view[i];
     return Emval.toHandle(a);
   },
+  _emval_array_to_memory_view__deps: ['$Emval'],
+  _emval_array_to_memory_view: (dst, src) => {
+    dst = Emval.toValue(dst);
+    src = Emval.toValue(src);
+    dst.set(src);
+  },
+#else
+  _emval_new_array_from_memory_view__deps: ['$Emval'],
+  _emval_new_array_from_memory_view: (view) => {
+    view = Emval.toValue(view);
+    var reader = (() => {
+      if (view.BYTES_PER_ELEMENT===1)
+        return i => view[i];
+      const dv = new DataView(view.buffer, view.byteOffset);
```

Comment:
Nit: you could get away with a lot less code by simply building a map from typed array names (since the list is exhaustive anyway) and avoiding custom reader wrappers. Something like:

```js
const dv = new DataView(view.buffer, view.byteOffset);
const reader = {
  Int8Array: dv.getInt8,
  Uint8Array: dv.getUint8,
  Int16Array: dv.getInt16,
  ...
}[view.constructor.name];
var a = new Array(view.length);
for (var i = 0; i < view.length; i++) a[i] = reader.call(dv, i * view.BYTES_PER_ELEMENT, true);
return Emval.toHandle(a);
```

Same for the writer below.

Reply:
I thought about something like that but I was not sure if this is expected to work classes that extend the typedarrays.

Reply:
I don't think we use anything like that. Extending typed arrays is very uncommon, but if we end up making one, we can add a prototype walk later.

Reply:
Oh right, you can use `view[Symbol.toStringTag]` instead. That one is preserved even in subclasses and is meant for differentiating between built-in objects.

Reply:
This has been implemented.